crates/pallet-bridge-pot-currency-swap/src/lib.rs

```diff
@@ -0,0 +1,126 @@
+//! A substrate pallet for bridge pot currency swap implementation.
+
+// Either generate code at stadard mode, or `no_std`, based on the `std` feature presence.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+    sp_runtime::traits::Convert,
+    sp_std::marker::PhantomData,
+    traits::{fungible::Inspect, Currency, StorageVersion},
+};
+pub mod existence_optional;
+pub mod existence_required;
+
+pub use existence_optional::Marker as ExistenceOptional;
+pub use existence_required::Marker as ExistenceRequired;
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+/// The interface to verify bridge pot currency swap related data at genesis.
+pub trait GenesisVerifier {
```

Comment:
I believe this crate should also offer a verifier implementation, and that this implementation should be properly tested, rather then mock-tested.

Reply:
I've created an issue - https://github.com/humanode-network/humanode/issues/699 to implement it in separate PR based on your review comment.